vendor/k8s.io/gengo/examples/deepcopy-gen/generators/deepcopy.go

```diff
@@ -184,10 +184,18 @@ func Packages(context *generator.Context, arguments *args.GeneratorArgs) generat
 
 		if pkgNeedsGeneration {
 			glog.V(3).Infof("Package %q needs generation", i)
+			path := pkg.Path
```

Comment:
Yeah, basically your "CanonicalizeImports" strips the information that gengo needs in order to output to the correct location.  I think there is a better fix in gengo, but I don't have the brainpower to refine it yet (and haven't seen all the cases).

Basically "k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/runtime" is a valid go path (to various tools), and also to the output base location that we need to write to.